datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -46,7 +51,10 @@ use super::ParquetFileMetrics;
 ///
 /// If an index IS present in the returned Vec it means the predicate
 /// did not filter out that row group.
+///
+/// Note: This method currently ignores ColumnOrder (#8342)
 pub(crate) fn prune_row_groups_by_statistics(
+    parquet_schema: &SchemaDescriptor,
```

Comment:
Tbh we need the full FileMetadata here in order to inspect the ColumnOrder - I decided against this as it would result in a load of test churn